builtin/game/chatcommands.lua

```diff
@@ -154,53 +154,72 @@ core.register_chatcommand("privs", {
 				core.get_player_privs(param), ' ')
 	end,
 })
+
+local function handle_grant_command(caller, grantname, grantprivstr)
+	if not core.check_player_privs(caller, {privs=true}) and
+			not core.check_player_privs(name, {basic_privs=true}) then
```

Comment:
Why not getting rid of `check_player_privs` here, when you're at it :-)

```lua
local privs = minetest.get_player_privs(caller)
if not privs.privilege_to_check and not privs.other_priv_to_check then
end
```

Furthermore i recommend, that `check_player_privs` needs to be deprecated.